bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/ScalarWebAvContentProtocol.java

```diff
@@ -0,0 +1,3226 @@
+                        "scalarwebavcontrolplsource", prefix + "Source", null));
+
+                descriptors.add(createDescriptor(createChannel(PL_STARTDATETIME, id, uri), "String",
+                        "scalarwebavcontrolplstartdatetime", prefix + "Start Date/Time", null));
+
+                descriptors.add(createDescriptor(createChannel(PL_TITLE, id, uri), "String",
+                        "scalarwebavcontrolpltitle", prefix + "Title", null));
+
+                descriptors.add(createDescriptor(createChannel(PL_TRIPLETSTR, id, uri), "String",
+                        "scalarwebavcontrolpltripletstr", prefix + "Triplet", null));
+
+                descriptors.add(createDescriptor(createChannel(PL_URI, id, uri), "String", "scalarwebavcontrolpluri",
+                        prefix + "URI", null));
+
+            }
+
+            if (VersionUtilities.equals(version, ScalarWebMethod.V1_2)) {
+                descriptors.add(createDescriptor(createChannel(PL_ALBUMNAME, id, uri), "String",
+                        "scalarwebavcontrolplalbumname", prefix + "Album Name", null));
+                descriptors.add(createDescriptor(createChannel(PL_APPLICATIONNAME, id, uri), "String",
+                        "scalarwebavcontrolplapplicationname", prefix + "Application Name", null));
+                descriptors.add(createDescriptor(createChannel(PL_ARTIST, id, uri), "String",
+                        "scalarwebavcontrolplartist", prefix + "Artist", null));
+                descriptors.add(createDescriptor(createChannel(PL_AUDIOCHANNEL, id, uri), "String",
+                        "scalarwebavcontrolplaudiochannel", prefix + "Audio Channel", null));
+                descriptors.add(createDescriptor(createChannel(PL_AUDIOCODEC, id, uri), "String",
+                        "scalarwebavcontrolplaudiocodec", prefix + "Audio Codec", null));
+                descriptors.add(createDescriptor(createChannel(PL_AUDIOFREQUENCY, id, uri), "String",
+                        "scalarwebavcontrolplaudiofrequency", prefix + "Audio Frequency", null));
+                descriptors.add(createDescriptor(createChannel(PL_BROADCASTFREQ, id, uri), "Number",
+                        "scalarwebavcontrolplbroadcastfreq", prefix + "Broadcast Frequency", null));
+                descriptors.add(createDescriptor(createChannel(PL_BROADCASTFREQBAND, id, uri), "String",
+                        "scalarwebavcontrolplbroadcastfreqband", prefix + "Broadcast Frequency Band", null));
+                descriptors.add(createDescriptor(createChannel(PL_CHANNELNAME, id, uri), "String",
+                        "scalarwebavcontrolplchannelname", prefix + "Channel Name", null));
+                descriptors.add(createDescriptor(createChannel(PL_CHAPTERCOUNT, id, uri), "Number",
+                        "scalarwebavcontrolplchaptercount", prefix + "Chapter Count", null));
+                descriptors.
```

Comment:
Line 2524:
Why bother checking this if you are justing going to go into a switch statement anyway?

Reply:
because the uri [pathpart(0)] is only valid on term (would be null on content).  I could potentially remove the check for content (since it's the final else) but I really don't want to do that for a few reasons - 1) if I add something to the bottom, I'd need to restore that if, 2) that's the pattern I follow across the setChannels in all protocols and I'd rather not vary from it